### PR TITLE
Initial macOS support

### PR DESCRIPTION
**Pull request by amirsaam**
As described in #24 
Conditionally Imports UIKit or AppKit based on OS
Added macOS in package for 10.11 (same release year as iOS9)
Added UIImage as only available in iOS and made NSImage available in macOS

### CODE REVIEW

File: DataCache.podspec [hunk 1]
```diff
@@ -6,6 +6,7 @@ Pod::Spec.new do |s|
   s.license      = { :type => 'MIT', :file => 'LICENSE' }
   s.source       = { :git => 'https://github.com/huynguyencong/DataCache.git', :tag => "#{s.version}" }
   s.author       = { 'Huy Nguyen Cong' => 'https://github.com/huynguyencong' }
+  s.osx.deployment_target = '10.11'
   s.ios.deployment_target = '13.0'
```
Comment (amirsaam) [inline DataCache.podspec:10]: is there a reason why swift package has ios9 but pod 13?
i don't know much about pod, so just wondered if i need to change osx target to match ios13 release date

Comment (huynguyencong) [reply DataCache.podspec:10]: It should be the same deployment target between Podspec and SPM

Comment (amirsaam) [reply DataCache.podspec:10]: So you want me to change it to ios9?

Comment (huynguyencong) [reply DataCache.podspec:10]: I think iOS 13 is ok. I will update it later.

**Comment by amirsaam** [comment]
Lemme know if you do want to merge this since I do want to use this package in [PlayCover](https://github.com/PlayCover/PlayCover)

**Comment by huynguyencong** [comment]
Have you tested it? If you tested it and it worked, I can merge it now then test it later.

**Comment by amirsaam** [comment]
I'll test it one more time to make sure everything works.
Last time (before this) I imported it and the only error was about UIKit and UIImage.

**Comment by amirsaam** [comment]
I just made a simple project to test.
you can simply run to check

**Comment by amirsaam** [comment]
Would be lovely to do a new tag 👍

**Comment by huynguyencong** [comment]
Thank for your contribution!

**Comment by amirsaam** [comment]
My pleasure!

**Pull request merged**